workspaces/spectacle/src/spectacle.ts

```diff
@@ -312,6 +355,23 @@ export async function makeSpectacle(opticContext: IOpticContext) {
       method: (parent: endpoints.RequestNodeWrapper) => {
         return Promise.resolve(parent.value.httpMethod);
       },
+      queries: async () => {
+        // TODO connect up to rust code
+        return [
+          {
+            id: 'queryid-1',
+            key: 'todoId',
+            rootShapeId: 'shape-1',
+            isRemoved: false,
+          },
+          {
+            id: 'queryid-2',
+            key: 'limit',
+            rootShapeId: 'shape-2',
+            isRemoved: false,
+          },
+        ];
```

Comment:
This isn't how the Optic Engine will be able to represent what the query parameters for a Request are, especially considering how we'll need to implement diffing. Instead of returning an array of query nodes, each with a `key` and it's own `rootShapeId`, each `Request` will have a single `Body` that represents the entire set of query parameters. This `Body` will point to a root shape which is always an object, with fields for each query parameter.

This approach is important because this allows us to leverage the existing Shape differ, since that just works on Bodies.

Reply:
Would this be an HTTPBody? If so, what would the contentType be (would this even be applicable)? Would we want to create our own type that doesn't include content type

I've updated this so it query returns a single HTTPBody

Reply:
Working on #937 has shown it's not _quite_ an `HTTPBody`, as there's no content type to consider. It's own type with only a `rootShapeId` and `isRemoved` seems appropriate.